Implement FAB for page specific actions

ZhangYiJiang opened the pull request:
This adds floating action buttons for page actions, which makes NUSMods easier to use on mobile (similar to proposal #411). The FAB comes from [Material Components Web](https://material.io/components/web/) using [RMWC](https://github.com/jamesmfriedman/rmwc) for React bindings. 

![screenshot from 2017-12-14 11-37-20](https://user-images.githubusercontent.com/445650/33974405-303b8310-e0c3-11e7-99e1-2fcc196fb60c.png) ![screenshot from 2017-12-14 11-37-31](https://user-images.githubusercontent.com/445650/33974406-307070fc-e0c3-11e7-8de2-2319af5be452.png)


Since the library is modular, we only need to import the components and styles we need, which makes it fairly lightweight. 

Page actions to implement: 

- [X] Open menu on module info page (we assume the most common action is browsing, not add/remove module from timetable)
- [x] Open filters list on module search page and venues page 
- [ ] Focus the module select box on timetable page (probably will implement after #484 goes through)

yangshun commented on v3/src/js/views/modules/ModulePageContent.scss:
```diff
@@ -1,9 +1,73 @@
+  top: $sticky-top;
+  margin-top: 0.4rem;
+
+  ul {
```
You can consider using Bootstrap's `list-unstyled` if it has been included in the main CSS bundle.

yangshun commented on .gitignore:
```diff
@@ -3,7 +3,7 @@
 .vagrant
 app/bower_components
-vendor
+# vendor
```
What's this change for?

ZhangYiJiang replied:
I was originally going to put the material styles under vendor, but I noticed the vendor folder was not properly checked in because this line was in the `.gitignore`. I think this line should probably be removed because it is not specific enough.

yangshun replied:
Ok please remove it then 😂

ZhangYiJiang commented:
Since the right hand sidebar on both module finder and module info work in the same manner, I've extracted the common parts into a `<SideMenu>` wrapper component which also contains the FAB acting as toggle. 

Because I couldn't fix the way the position sticky search box and position fixed drawer menu interacted with each other, I moved `<CorsNotification>` from `<AppShell>` to the timetable and module info page instead. This does mean that the warning will reappear on every page change on desktop, but I think it's okay. 

re: using RMWC - I intend to use the Snackbar component for #505

ZhangYiJiang commented:
@taneliang That is a little odd. It's almost certainly an upstream bug - Gmail on Android does not have the same behaviour. I'm guessing what's happening is that the long press is getting converted into a right click on touch screens. It's not a huge issue anyway, because I don't see people trying to long press FABs.
